src/parser/src/Type.cpp

```diff
@@ -29,7 +29,7 @@ int Typing::MatrixType::getLength() const {
     return std::accumulate(this->dimensions.begin(), this->dimensions.end(), 1, std::multiplies());
 }
 llvm::Type* Typing::MatrixType::getLLVMType(Utils::IRContext* context) const {
-    llvm::Type* ty;
+    llvm::Type* ty = nullptr;
```

Comment:
I personally prefer the uniform initialiser syntax `llvm::Type* ty {};`, but comes to the same effect